internal/shared/kubernetes/kubernetes.go

```diff
@@ -15,3 +20,59 @@ func IsValidKubernetesNamespace(name string) bool {
 	}
 	return false
 }
+
+// GetGitSyncInstanceAnnotation returns the application instance name from annotation
+func GetGitSyncInstanceAnnotation(un *unstructured.Unstructured, key string) (string, error) {
+	annotations, err := nestedNullableStringMap(un.Object, "metadata", "annotations")
+	if err != nil {
+		return "", fmt.Errorf("failed to get annotations from target object %s %s/%s: %w", un.GroupVersionKind().String(), un.GetNamespace(), un.GetName(), err)
+	}
+	if annotations != nil {
+		return annotations[key], nil
+	}
+	return "", nil
+}
+
+// SetGitSyncInstanceAnnotation the recommended app.kubernetes.io/instance annotation against an unstructured object
```

Comment:
```suggestion
// SetGitSyncInstanceAnnotation sets the recommended app.kubernetes.io/instance annotation against an unstructured object
```

Reply:
Regarding the comment on "legacy labeling if environment variable is set" below - want to make sure that it is the case.?

Reply:
No, removed.